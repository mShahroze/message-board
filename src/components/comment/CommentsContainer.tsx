import React, { useEffect, useState } from 'react';
import fetchComments from '../../services/api';
import { Comment } from '../../types/Comment';
import SingleComment from './Comment';

const CommentsContainer = (): JSX.Element => {
  const [comments, setComments] = useState<Comment[]>();

  useEffect(() => {
    async function getData() {
      const { data } = await fetchComments();
      setComments(data);
    }

    getData();
  }, []);

  return comments ? (
    <div className='comments'>
      {comments.map((comment) => {
        return <SingleComment comment={comment} key={comment.id} />;
      })}
    </div>
  ) : (
    <div></div>
  );
};

export default CommentsContainer;
