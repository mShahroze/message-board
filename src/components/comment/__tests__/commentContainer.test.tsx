/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { act, render } from '@testing-library/react';
import fetchMock from 'jest-fetch-mock';
import commentsMockData from '../../../testData/commentsData.json';
import CommentsContainer from '../CommentsContainer';

describe('CommentContainer Component', () => {
  it('should take a snapshot', async () => {
    fetchMock.mockResponseOnce(JSON.stringify(commentsMockData));
    let component: any;
    await act(async () => {
      component = render(<CommentsContainer />);
    });
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });
  it('No Data received', async () => {
    fetchMock.mockResponseOnce(JSON.stringify({ data: [] }));
    let component: any;
    await act(async () => {
      component = render(<CommentsContainer />);
    });
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });
});
