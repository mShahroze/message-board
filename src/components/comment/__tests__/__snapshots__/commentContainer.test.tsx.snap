// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentContainer Component No Data received 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`CommentContainer Component should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="comment"
    >
      <p>
        Could someone point out to me, what the benefit of winning the group is? I mean, both finalists of the group will qualify for the Spring Finals? So what is the point of fighting for 1st? Bragginâ€™ rights?
      </p>
      <span>
        MakeNBreak69
      </span>
      <div
        data-testid="comment-0001-comment"
      >
        View 2 Replies
      </div>
    </div>
    <div
      class="comment"
    >
      <p>
        NiP have the stream running on a big TV in the same room they're playing in.Are we doing this whole shenanigans again now? Didn't ESIC say there are gonna be consequences from now on? Getting a little tired of this.
      </p>
      <span>
        z0ttel89
      </span>
      <div
        data-testid="comment-0003-comment"
      >
        View 2 Replies
      </div>
    </div>
    <div
      class="comment"
    >
      <p>
        Blast has the best presentation and production of all the ongoing series at the moment, my favourite without a doubt. That sounded like a paid advertisement, but I really just like the series
      </p>
      <span>
        Spartan_exr
      </span>
      0
    </div>
  </div>
</DocumentFragment>
`;
